statsmodels/tools/decorators.py

```diff
@@ -4,7 +4,26 @@
 from statsmodels.compat.python import get_function_name
 import warnings
 
-__all__ = ['resettable_cache', 'cache_readonly', 'cache_writable']
+__all__ = ['resettable_cache', 'cache_readonly', 'cache_writable',
+           'deprecated_alias']
+
+
+def deprecated_alias(old_name, new_name, remove_version=None):
+    msg = '%s is a deprecated alias for %s' % (old_name, new_name)
+    if remove_version is not None:
+        msg += ', will be removed in version %s' % remove_version
+
+    def fget(self):
+        warnings.warn(msg)
+        return getattr(self, new_name)
+
+    def fset(self, value):
+        warnings.warn(msg)
```

Comment:
does this need a warnings class, e.g. DeprecationWarning.
what kind of warn does numpy deprecate issue?

Reply:
Looks like np.deprecate does `warnings.warn(depdoc, DeprecationWarning, stacklevel=2)`.  This is easy to implement, a bit harder to test.  I'll take a look for examples elsewhere.